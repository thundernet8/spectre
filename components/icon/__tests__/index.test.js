import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import Icon from "..";

describe("Icon", () => {
    it("should render Icon", () => {
        const icon = shallow(<Icon type="upward" />);
        expect(icon.type()).toEqual("i");
        expect(icon.hasClass("spticon")).toBe(true);
        expect(icon.hasClass("spticon-upward")).toBe(true);

        const spinIcon = shallow(<Icon type="upward" spin />);
        expect(spinIcon.hasClass("spticon-spin")).toBe(true);

        const faIcon = shallow(<Icon fontset="fa" type="home" />);
        expect(faIcon.hasClass("fa fa-home")).toBe(true);
        expect(faIcon.hasClass("spticon-home")).toBe(false);
    });

    it("icon renders correctly", () => {
        const tree = renderer.create(<Icon type="upward" />).toJSON();
        expect(tree).toMatchSnapshot();

        const tree2 = renderer
            .create(<Icon fontset="fa" type="home" />)
            .toJSON();
        expect(tree2).toMatchSnapshot();
    });
});
