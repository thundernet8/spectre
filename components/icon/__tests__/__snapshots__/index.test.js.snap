// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon icon renders correctly 1`] = `
<i
  className="icon spticon spticon-upward"
/>
`;

exports[`Icon icon renders correctly 2`] = `
<i
  className="icon fa fa-home"
  fontset="fa"
/>
`;
